### ISSUE TITLE
Instructions on how to run on arbitrary football videos?

### ISSUE THREAD

**Issue by trekze** [cioppaanthony/context-aware-loss#4]
Hi there. 

First of all, amazing project!

I'm trying to run what you've built on some non-SoccerNet footage, but I'm not getting results that make sense. I was hoping you could confirm that I'm taking the right steps:

- Downsample the videos to 2fps (tried 1080p and 244p)
- Run Resnet-152 feature extraction as described here:
 https://github.com/SilvioGiancola/SoccerNet-code/tree/master/src/feature_extraction
- Run PCA512 on the features:
```
pca = PCA(n_components=512)
npy = np.load(os.path.join(gameDir, "1_ResNET.npy"))
result = pca.fit_transform(npy)
```
- Run the result through your code by:
  
1. placing our data where `listgame_Test_100_ResNET_PCA512_features.npy` would be normally
2. making some slight modifications to your code to remove the need for labels / mAP calculation

It all runs, and the resulting segmentation/detection graph looks like it could be correct, with peaks and valleys, but it doesn't match what we see in the footage. Any idea which step above could be incorrect?

Thank you!

**Comment by cioppaanthony** [comment]
Hi,

Thanks for your interest in our work ! 

Actually, we've known this issue for quite a while now. This is due to how the ResNET features and the PCA were computed to first extract the features provided by SoccerNet. The PCA is actually something else than:

<code>pca = PCA(n_components=512)</code>

which results in a different distribution of the features (but we unfortunately don't have the exact code). Therefore, If you train the model with what is provided with SoccerNet and try to test it on an arbitrary video with your features, the network won't be able to do a correct spotting of the actions since the distribution of these features is different than what was used for the training.

To solve this issue, we are currently in touch with the author of SoccerNet to release a reproducible code for extracting the features so that anyone can easily extract them on any videos and use our pre-trained weights.

In the meantime, a solution is to extract all features from the videos of SoccerNet with the method that you describe and re-train a network on them. Then, you should be able to test it on any video with the same feature extractor.

I hope this helps for now.

**Comment by trekze** [comment]
Hi @cioppaanthony 

Thank you for your detailed response. Good to know that we were on the right track. We will consider re-training the network using our own feature extraction. Meanwhile if you do get a response from SoccerNet, do let us know, with a mention here, as that would a preferable alternative to re-training. Keep up the good work.

**Comment by cioppaanthony** [comment]
Hi,

Sure, I will leave this issue opened until we get a reproducible way to compute the features for SoccerNet, and tag you when I close it.

**Comment by NikkiGaze** [comment]
Good day, Anthony.
Great work, thank you!
I'm trying to reproduce your results and stucked into the same problem. Seems like solution above can help, but may be there are some news about PCA extraction used by you?

**Comment by cioppaanthony** [comment]
Hi @NikkiGaze and @hmexx ,

Thanks for your interest ! I'm coming with great news for you. :-)

We have worked on that issue and now released a reproducible way to get the features from the video! It is part of our new dataset called SoccerNet-v2 (which extends the annotations of SoccerNet to 17 classes and includes two extra tasks). 

This has been recently released on a new repository: https://github.com/SilvioGiancola/SoccerNetv2-DevKit

My advice would be to migrate to this new repository as it contains everything you need:
- Function to extract the features: https://github.com/SilvioGiancola/SoccerNetv2-DevKit/tree/main/Features
- Functions to train or infer with CALF: https://github.com/SilvioGiancola/SoccerNetv2-DevKit/tree/main/Task1-ActionSpotting/CALF

Also, we re-implemented the action spotting network in Pytorch and very much optimized the training process of the network as well if you wish to also train it yourself. Instead of taking 12-24 hours, it now only takes about 2-3 hours to train on the whole dataset.

I hope this will help you in your research. Otherwise, don't hesitate to contact me on the new repository via an issue, by email, or directly on our new discord for the challenge on SoccerNet-v2 for a live chat:  https://discord.gg/m5w9g4kE34​.

Have a great day.
